trollmoves/filescleaner.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025 Pytroll Developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Utility functions for cleaning files and directories."""
+
+
+import datetime as dt
+import logging
+import os
+from glob import glob
+
+LOGGER = logging.getLogger("__name__")
+
+
+try:
+    from posttroll.message import Message
+except ImportError:
+    def Message(*args, **kwargs):
+        """Fake message object in case posttroll isn't installed."""
+        del args, kwargs
+
+
+class FilesCleaner():
+    """Files cleaner class to accomodate cleaning of files acording to configured rules."""
+
+    def __init__(self, publisher, section, info, dry_run=True):
+        """Initialize the class."""
+        self.pub = publisher
+        self.section = section
+        self.info = info
+        self.dry_run = dry_run
+        self.st_time = self.info.get('st_time', 'st_ctime')
```

Comment:
```suggestion
        self.stat_time_method = self.info.get('st_time', 'st_ctime')
```

I think the attribute name could be more descriptive. My suggestion might not be much better, though :sweat_smile: 

There is no `st_time` in the output of `os.lstat()`. Did you mean either `st_atime` or `st_mtime`?

Reply:
Ok, thanks. Yes, maybe your suggestion is better. I will contemplate a bit more and decide... Perhaps others have thoughts?